applications/serial_lte_modem/src/twi/slm_at_twi.c

```diff
@@ -0,0 +1,404 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <logging/log.h>
+#include <zephyr.h>
+#include <drivers/i2c.h>
+#include <stdio.h>
+#include "slm_util.h"
+#include "slm_at_twi.h"
+
+LOG_MODULE_REGISTER(slm_twi, CONFIG_SLM_LOG_LEVEL);
+
+#define TWI_MAX_INSTANCE	4
+#define TWI_ADDR_LEN		2
+#define TWI_DATA_LEN		255
+
+/**@brief List of supported AT commands. */
+enum slm_twi_at_cmd_type {
+	AT_TWI_OPEN,
+	AT_TWI_WRITE,
+	AT_TWI_READ,
+	AT_TWI_WRITE_READ,
+	AT_TWI_MAX
+};
+
+static const struct device *slm_twi_dev[TWI_MAX_INSTANCE];
+
+/* global functions defined in different files */
+void rsp_send(const uint8_t *str, size_t len);
+
+/* global variable defined in different files */
+extern struct at_param_list at_param_list;
+extern char rsp_buf[CONFIG_SLM_SOCKET_RX_MAX * 2];
+
+static int do_twi_open(uint16_t idx)
+{
+	int err = -EINVAL;
+
+	switch (idx) {
+	case 0:
+		slm_twi_dev[idx] = device_get_binding(DT_LABEL(DT_NODELABEL(i2c0)));
+		break;
+	case 1:
+		slm_twi_dev[idx] = device_get_binding(DT_LABEL(DT_NODELABEL(i2c1)));
+		break;
+	case 2:
+		slm_twi_dev[idx] = device_get_binding(DT_LABEL(DT_NODELABEL(i2c2)));
+		break;
+	case 3:
+		slm_twi_dev[idx] = device_get_binding(DT_LABEL(DT_NODELABEL(i2c3)));
+		break;
+	default:
+		return err;
+	}
+
+	if (!slm_twi_dev[idx]) {
+		LOG_ERR("Could not get pointer to device");
+		return err;
+	}
+
+	return 0;
+}
+
+static int do_twi_write(uint16_t index, uint16_t dev_addr,
+			const uint8_t *twi_data_ascii, uint16_t ascii_len)
+{
+	int ret = -EINVAL;
+	uint8_t twi_data_hex[ascii_len/2];
+
+	if (!slm_twi_dev[index]) {
+		LOG_ERR("TWI device is not opened");
+		return ret;
+	}
+
+	ret = slm_util_atoh(twi_data_ascii, ascii_len, twi_data_hex, ascii_len/2);
```

Comment:
```suggestion
	ret = slm_util_atoh(twi_data_ascii, ascii_len, twi_data_hex, ascii_len / 2);
```
In general Zephyr coding style requires a space around most of the operators, please fix in other places below as well.